src/Engine/Objects/SpriteObject.cpp

```diff
@@ -867,29 +866,28 @@ void SpriteObject::InitializeSpriteObjects() {
     }
 }
 
-void SpriteObject::_46BEF1_apply_spells_aoe() {
-    __debugbreak();  // Ritor1
-
-    for (size_t i = 0; i < pActors.size(); ++i) {
-        if (pActors[i].CanAct()) {
-            int v7 = pActors[i].vPosition.x - this->vPosition.x;
-            int v9 = pActors[i].vPosition.y - this->vPosition.y;
-            int v10 = pActors[i].uActorHeight / 2 + pActors[i].vPosition.z -
-                      this->vVelocity.y;
-
-            int v11 = this->vVelocity.x * this->vVelocity.x;
-
-            if (v11 >= v7 * v7 + v9 * v9 + v10 * v10) {
-                if (pActors[i].DoesDmgTypeDoDamage(DMGT_DARK)) {
-                    pActors[i].pActorBuffs[ACTOR_BUFF_INDEX(this->uSpellID)].Apply(
-                        GameTime(pParty->GetPlayingTime() +
-                            GameTime::FromSeconds(this->spell_level)),
-                        this->spell_skill, 4, 0, 0);
-                    pActors[i].uAttributes |= ACTOR_AGGRESSOR;
+bool SpriteObject::applyShrinkRayAoe() {
+    bool isApplied = false;
+    // Calculation was moved from initial sprite creation processing
+    GameTime duration = GameTime::FromMinutes(this->spell_level * 5);
+    static const int shrinkPower = 4;
+
+    for (Actor &actor : pActors) {
+        // TODO(Nik-RE-dev): paralyzed actor will not be affected?
+        if (actor.CanAct()) {
+            int distanceSq = (actor.vPosition - this->vPosition + Vec3i(0, 0, actor.uActorHeight / 2)).LengthSqr();
+            int checkDistanceSq = (256 + actor.uActorRadius) * (256 + actor.uActorRadius);
```

Comment:
Could you add this radius to config please

Reply:
Would it be okay to add it in next PR?
I have other changes in the ready (waiting for this PR) that refactor AOE queue as well as add config value for this and for damaging AOE spells radius.

Reply:
Sure